sfge/src/main/java/com/salesforce/rules/ops/boundary/SharingPolicyBoundary.java

```diff
@@ -0,0 +1,130 @@
+package com.salesforce.rules.ops.boundary;
+
+import com.salesforce.graph.Schema;
+import com.salesforce.graph.vertex.SFVertex;
+import com.salesforce.graph.vertex.SFVertexFactory;
+import com.salesforce.graph.vertex.SharingPolicy;
+import com.salesforce.graph.vertex.UserClassVertex;
+import java.util.List;
+import java.util.Optional;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+
+/**
+ * This class stores a {@link UserClassVertex} boundary item and can try to find the nearest
+ * ancestor with an explicitly defined sharing policy (the "applicable ancestor"). See {@link
+ * #getApplicableAncestorPolicyVertex()}.
+ */
+public class SharingPolicyBoundary implements Boundary<SFVertex> {
```

Comment:
Since `getBoundaryItem` returns a `UserClassVertex`, this should be `Boundary<UserClassVertex>`.

Reply:
I don't quite understand this, but changing this results in an error within `SharingPolicyBoundaryDetector` that says:
> Type parameter 'com.salesforce.rules.ops.boundary.SharingPolicyBoundary' is not within its bound; should implement 'com.salesforce.rules.ops.boundary.Boundary<com.salesforce.graph.vertex.SFVertex>'

I'm a bit confused since I thought `UserClassVertex` implements `InheritableSFVertex` (which implements `SFVertex`). Do you know what this means? I'm not super well-versed in Java generics.